sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ModelsRepositoryClient.cs

```diff
@@ -120,27 +124,29 @@ public ModelsRepositoryClient(Uri repositoryUri, ModelsRepositoryClientOptions o
         }
 
         /// <summary>
-        /// Resolves a collection of model definitions identified by <paramref name="dtmis"/> and optionally their dependencies.
+        /// Gets a collection of model definitions identified by <paramref name="dtmis"/> and optionally their dependencies.
         /// </summary>
         /// <returns>
         /// An IDictionary containing the model definition(s) where the key is the dtmi
         /// and the value is the raw model definition string.
         /// </returns>
         /// <exception cref="RequestFailedException">Thrown when a resolution failure occurs.</exception>
         /// <param name="dtmis">A collection of well-formed DTDL model Ids.</param>
+        /// <param name="resolutionOption">A DependencyResolutionOption value to force model resolution behavior.</param>
         /// <param name="cancellationToken">The cancellationToken.</param>
         [SuppressMessage(
             "Usage",
             "AZC0015:Unexpected client method return type.",
             Justification = "Item lookup is optimized with a dictionary type, we do not expect any more than ~20 items to be returned. TODO: azabbasi: discuss this issue with the review board.")]
-        public virtual async Task<IDictionary<string, string>> ResolveAsync(IEnumerable<string> dtmis, CancellationToken cancellationToken = default)
+        public virtual async Task<IDictionary<string, string>> GetModelsAsync(
+            IEnumerable<string> dtmis, DependencyResolutionOption? resolutionOption = null, CancellationToken cancellationToken = default)
         {
-            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ModelsRepositoryClient)}.{nameof(Resolve)}");
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ModelsRepositoryClient)}.{nameof(GetModels)}");
```

Comment:
GetModelsAsync

Reply:
Scopes generally don't have an async suffix. that's also the case in the generated code